### PR TITLE
Shopping cart medium solution added.

### PR DESCRIPTION
**Pull request by garimasingh128**

### CODE REVIEW

**Review by girlscript-blr — CHANGES_REQUESTED** [review]
There are few bugs in your solution as follows:
- If I enter my name with a space, it takes phone number as my second name and doesn't allow me to fill phone number.
Screenshot 2020-07-08 at 12 04 08 AM

- There is a mismatch between item numbers and item getting displayed. Eg: When I select 4 I should get Bolt earphones but Bluetooth computer mouse is getting selected. 
- For 5. Boat earphones the discount price is more than the original price. User shouldn't pay in negative :p 
- Rename Money you save to Amount to be paid/ Final/ Discounted Amount.
Screenshot 2020-07-08 at 12 06 00 AM
- In the bill, invoice Quantity is getting displayed as 0
- The 6% tax getting calculated seems to be wrong, as 6% of 5600 is coming to be 336.
- For Home delivery, extra amount to be paid is not taken into consideration. Eg: I selected 23kms, so extra Rs.30 needs to be added as shipping charges.
- `Amount Saved by you` & `Amount to be Paid` seems to be displayed wrongly as per the meaning.
Screenshot 2020-07-08 at 12 11 18 AM

**Comment by garimasingh128** [comment]
I have updated the code. Please check!

File: Medium/1. Shopping Cart Problem/solutions/garimasingh128_Garima Singh/main.cpp [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		tax = 0.06 * price;
+	total = tax + price;
+	if(p==2)
+	{
+		if(dist<=5)
+		total=total;
+		if(dist>5 && dist<=20)
+		total=total+30;
+		if(dist>20 && dist<=50)
+		total=total+90;
```
Comment (girlscript-blr) [inline Medium/1. Shopping Cart Problem/solutions/garimasingh128_Garima Singh/main.cpp]: For distance from 20kms to 50kms we are charging delivery charges as 60 not 90.

**Review by girlscript-blr — CHANGES_REQUESTED** [review]
There is still a logical error in totaling the amount. As per the items I selected to buy i.e.
`Bolt Earphones  3  => 280*3 = 840`
`Bluetooth Mouse 2 => 400*2 = 800`
`Basehead earphones 1 => 1000`
`Total = 840+800+1000 = 2640`
`Tax = 0.06*2640 = 158.4`
`Distance 25kms = 90(as per your solution, please change it to 60)`
`Total Amount to be paid = 2640 +158.4 +90 = 2888.4`

But as per your solution, it is **5072** (Please don't make the customer sue our shop :p )
Screenshot 2020-07-08 at 9 08 37 PM

File: Medium/1. Shopping Cart Problem/solutions/garimasingh128_Garima Singh/main.cpp (hunk 1 above)
Comment (garimasingh128) [reply Medium/1. Shopping Cart Problem/solutions/garimasingh128_Garima Singh/main.cpp]: I though cumulative amount is to be taken into count!

**Comment by garimasingh128** [comment]
I hope now the issue is fixed!

**Review by girlscript-blr — APPROVED** [review]

**Comment by girlscript-blr** [comment]
Greetings from Code With Girlscript Bangalore😃

Dear participant, we have gone through your pull requests made in `MEDIUM` sections of the problem statement.

Congratulations🎊 🎉  on getting your pull request successfully merged in the `Shopping Cart Problem - Medium` section in `code-with-girlscript-bangalore` repository. We will be distributing your valuable badges🥇🥈🥉🎖️ corresponding to the pull requests that are merged.

Learning is a never-ending process. 📈 Check out the [Advanced](https://github.com/girlscript-blr/code-with-girlscript-bangalore/tree/master/Hard/1.%20Shopping%20Cart%20Problem) of the problem and try your hand at it to earn **THE AFICIONADO** 🥇 and **GEEK OF THE GEEKS** 🏆 badges. Thanks for your contribution! Happy coding😃

**Pull request merged**